Editor/EditorCore/ShapeComponentEditor.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.ProBuilder.Shapes;
+using UnityEngine.UIElements;
+
+#if UNITY_2020_2_OR_NEWER
+using ToolManager = UnityEditor.EditorTools.ToolManager;
+#else
+using ToolManager = UnityEditor.EditorTools.EditorTools;
+#endif
+
+namespace UnityEditor.ProBuilder
+{
+    [CustomEditor(typeof(ShapeComponent))]
+    class ShapeComponentEditor : Editor
+    {
+        ShapeComponent m_ShapeComponent;
+        IMGUIContainer m_ShapeField;
+
+        SerializedProperty m_shape;
+        string[] m_ShapeTypes;
+        Type[] m_AvailableShapeTypes;
+
+        int m_ActiveShapeIndex = 0;
+
+        const string k_dialogTitle = "Shape reset";
+        const string k_dialogText = "The current shape has been edited, you will loose all modifications.";
+
+        ShapeComponentEditor()
+        {
+            m_AvailableShapeTypes =  TypeCache.GetTypesWithAttribute<ShapeAttribute>().Where(t => t.BaseType == typeof(Shape)).ToArray();
```

Comment:
Looks like this would break if there is an abstract implementation of shape, ex:

```
abstract class ProceduralShape : Shape {}
class ProceduralCube : ProceduralShape {}
```